client/jetpack-onboarding/steps/business-address.jsx

```diff
@@ -48,13 +49,14 @@ class JetpackOnboardingBusinessAddressStep extends React.PureComponent {
 		};
 	}
 
-	handleAddBusinessAddress = () => {
+	handleSubmit = () => {
 		const { siteId } = this.props;
 		this.props.saveJetpackOnboardingSettings( siteId, { businessAddress: this.state } );
+		page( this.props.getForwardUrl() );
```

Comment:
I think you should use `page.redirect` and have a `event.preventDefault()` as well (as described also in #21382).